utils/clickhouse.cpp

```diff
@@ -0,0 +1,88 @@
+/**
+ *		Tempesta FW
+ *
+ * Copyright (C) 2024 Tempesta Technologies, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License,
+ * or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program; if not, write to the Free Software Foundation, Inc., 59
+ * Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+ */
+
+#include "clickhouse.h"
+#include <iostream>
+#include <clickhouse/client.h>
+#include <clickhouse/base/socket.h>
+
+using namespace std;
+using namespace chrono;
+
+#define MAX_MSEC 100
+#define MAX_EVENTS 1000
+
+#define NOW_MS() \
+	duration_cast<milliseconds>(system_clock::now().time_since_epoch()).count()
+
+
+TfwClickhouse::TfwClickhouse(string host, string table_name,
+			     unsigned int cpu_cnt, Block *(*cb)())
+{
+	unsigned int i;
+
+	block_callback = cb;
+	this->table_name = table_name;
+	this->cpu_cnt = cpu_cnt;
+
+	tasks = new TfwTask[cpu_cnt];
```

Comment:
I think we should handle a case when `new` fails. We should check that if `tasks` == NULL and throw exception.